runtime/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt

```diff
@@ -147,99 +117,143 @@ public class JsonObjectBuilder internal constructor() {
 
     /**
-     * Add the given JSON [element] to a resulting array.
+     * Adds the given JSON [element] to a resulting array.
+     *
+     * Always returns `true` similarly to [ArrayList] specification.
      */
-    public fun add(element: JsonElement) {
+    public fun add(element: JsonElement): Boolean {
         content += element
+        return true
     }
 
     internal fun build(): JsonArray = JsonArray(content)
 
     /**
      * Adds [this] value to the current [JsonArray] as [JsonPrimitive].
      */
-    @Deprecated(message = unaryPlusDeprecated, replaceWith = ReplaceWith("add(this)"))
+    @Deprecated(message = unaryPlusDeprecated, replaceWith = ReplaceWith("add(this)"), level = DeprecationLevel.ERROR)
     public operator fun String?.unaryPlus() {
         content.add(JsonPrimitive(this))
     }
 
     /**
      * Adds [this] value to the current [JsonArray] as [JsonPrimitive].
      */
-    @Deprecated(message = unaryPlusDeprecated, replaceWith = ReplaceWith("add(this)"))
+    @Deprecated(message = unaryPlusDeprecated, replaceWith = ReplaceWith("add(this)"), level = DeprecationLevel.ERROR)
     public operator fun Number?.unaryPlus() {
         content.add(JsonPrimitive(this))
     }
 
     /**
      * Adds [this] value to the current [JsonArray] as [JsonPrimitive].
      */
-    @Deprecated(message = unaryPlusDeprecated, replaceWith = ReplaceWith("add(this)"))
+    @Deprecated(message = unaryPlusDeprecated, replaceWith = ReplaceWith("add(this)"), level = DeprecationLevel.ERROR)
     public operator fun Boolean?.unaryPlus() {
         content.add(JsonPrimitive(this))
     }
 
     /**
      * Adds [this] value to the current [JsonArray].
      */
-    @Deprecated(message = unaryPlusDeprecated, replaceWith = ReplaceWith("add(this)"))
+    @Deprecated(message = unaryPlusDeprecated, replaceWith = ReplaceWith("add(this)"), level = DeprecationLevel.ERROR)
     public operator fun JsonElement.unaryPlus() {
         this@JsonArrayBuilder.content.add(this)
     }
 }
 
+/**
+ * Adds the given boolean [value] to a resulting array.
+ *
+ * Always returns `true` similarly to [ArrayList] specification.
+ */
+public fun JsonArrayBuilder.add(value: Boolean?): Boolean = add(JsonPrimitive(value))
+
+/**
+ * Adds the given numeric [value] to a resulting array.
+ *
+ * Always returns `true` similarly to [ArrayList] specification.
+ */
+public fun JsonArrayBuilder.add(value: Number?): Boolean = add(JsonPrimitive(value))
+
+/**
+ * Adds the given string [value] to a resulting array.
+ *
+ * Always returns `true` similarly to [ArrayList] specification.
+ */
+public fun JsonArrayBuilder.add(value: String?): Boolean = add(JsonPrimitive(value))
+
+/**
+ * Adds the [JSON][JsonObject] produced by the [builderAction] function to a resulting array.
+ *
+ * Always returns `true` similarly to [ArrayList] specification.
+ */
+public fun JsonArrayBuilder.putJsonObject(builderAction: JsonObjectBuilder.() -> Unit): Boolean =
```

Comment:
should be `add`

Reply:
sorry, my bad, should have checked the result of rename refactoring